Refactors lib/private/Accounts classes

fsamapoor opened the pull request:
Following [previous PRs taking advantage of PHP8's constructor property promotion](https://github.com/nextcloud/server/pulls?q=is%3Apr+author%3Afsamapoor+Uses+PHP8%27s+constructor+property+promotion) in `/core/` namespace, I have also made the required adjustments to the classes in `/lib/private/Accounts` namespace.

The improvements in this PRs include but are not limited to:
- Using PHP8's constructor property promotion
- Adding return types
- Adding types to properties
- Removing redundant docblocks

fsamapoor committed:
Refactors lib/private/Accounts.

 Mainly using PHP8's constructor property promotion.

Signed-off-by: Faraz Samapoor <fsa@adlas.at>

fsamapoor committed:
Fixes Psalm error.

Signed-off-by: Faraz Samapoor <fsa@adlas.at>

nfebe reviewed (approved):
Looks okay! +1 on the more precise match.

Thank you

solracsf commented:
CI is failing.

Commit:
Fixes testcase errors.

Signed-off-by: Faraz Samapoor <fsa@adlas.at>

fsamapoor commented:
@solracsf Thank you for the heads up. I wasn't aware that the failing CI was in fact a result of failing test cases. I have fixed the failing tests and everything seems to be fine now.

blizzz commented on lib/private/Accounts/AccountManager.php:
```diff
@@ -138,35 +107,22 @@ class AccountManager implements IAccountManager {
+		private IVerificationToken $verificationToken,
+		private IMailer $mailer,
+		private Defaults $defaults,
+		private IFactory $factory,
```
```suggestion
		private IFactory $l10nFactory,
```

rename this for readbility/clarity, factory alone is a bit broad

fsamapoor replied:
@blizzz Thanks. Other than making it more readable, I searched the codebase and almost all other classes also used this convention. So keeping it consistent is always good.

blizzz reviewed (changes requested):
small nitpick

fsamapoor committed:
Renames the property to improve the code readbility/clarity.

Based on:
 https://github.com/nextcloud/server/pull/38978#pullrequestreview-1514202944

Co-authored-by: Arthur Schiwon <blizzz@arthur-schiwon.de>
Signed-off-by: Faraz Samapoor <fsa@adlas.at>

blizzz commented:
failing tests unrelated => merging
